propolis/src/hw/pci/mod.rs

```diff
@@ -8,26 +9,63 @@ use crate::common::*;
 use crate::dispatch::DispCtx;
 use crate::intr_pins::IntrPin;
 
+use num_enum::TryFromPrimitive;
+
+pub mod bar;
 pub mod bits;
+pub mod bus;
 mod device;
 
+pub use bus::Bus;
 pub use device::*;
 
-pub const PORT_PCI_CONFIG_ADDR: u16 = 0xcf8;
-pub const PORT_PCI_CONFIG_DATA: u16 = 0xcfc;
-
-const MASK_FUNC: u8 = 0x07;
-const MASK_DEV: u8 = 0x1f;
-const MASK_BUS: u8 = 0xff;
+#[derive(Copy, Clone, Eq, PartialEq, Debug, Ord, PartialOrd)]
+pub struct BusNum(u8);
+impl BusNum {
+    pub const fn new(n: u8) -> Option<Self> {
```

Comment:
Does this also return Option just for consistency with`DevNum` and `FuncNum` or will there be cases where it might fail?

Reply:
For consistency only